### ISSUE TITLE
CKEditor 33.0.0 List plugin not compatible with Drupal Media

### ISSUE THREAD

**Issue by lauriii** [ckeditor/ckeditor5#11490]
## 📝 Provide detailed reproduction steps (if any)
Drupal Media has following conversion:

Model:
```
<drupalMedia><caption>...</caption></drupalMedia>
```

Data:
```
<drupal-media data-caption></drupal-media>
```

This is not compatible with the list plugin. It could be that Drupal Media was never compatible with the List plugin but for some reason the `mapping-view-position-parent-not-found` exception was not thrown prior to 33.0.0. This seems to be caused by the `<caption>` not having a view element that it could be mapped to.

It could be possible to reproduce this with the `should allow nesting downcast conversion` CKEditor test when list plugin is enabled.

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by wimleers** [comment]
Sibling Drupal.org issue: https://www.drupal.org/project/drupal/issues/3269619

**Comment by niegowski** [comment]
This is happening because of the change released in the 33.0.0 release: [https://ckeditor.com/docs/ckeditor5/latest/updating/migration-to-33.html#responsibility-shift-in-the-lowlevel-downcast-converters](https://ckeditor.com/docs/ckeditor5/latest/updating/migration-to-33.html#responsibility-shift-in-the-lowlevel-downcast-converters)

As I can see, the [modelViewSplitOnInsert](https://github.com/ckeditor/ckeditor5/blob/033ca400cc1320a951b23e14ddcb43c9c68c71fe/packages/ckeditor5-list/src/list/converters.js#L220) downcast callback, registered by the link feature, does not verify if the conversion item was already consumed.

I also verified whether we can use the `editor.data.stringify( captionElement )` inside a downcast converter, and unfortunately, it's not that easy yet. There are 2 things that should be resolved to be able to trigger it instead of firing downcast dispatcher events: 1. the mapper is cleared on event `DataController.toView()` call; 2. the consumables consumed in the nested stringify call are not consumed in the outer conversion call. 

I'll prepare a PR that fixes the [modelViewSplitOnInsert](https://github.com/ckeditor/ckeditor5/blob/033ca400cc1320a951b23e14ddcb43c9c68c71fe/packages/ckeditor5-list/src/list/converters.js#L220) conversion callback so it won't try to convert anything if the element was already consumed.
